src/main/java/ro/unibuc/triplea/application/auth/web/GameController.java

```diff
@@ -0,0 +1,49 @@
+package ro.unibuc.triplea.application.auth.web;
+
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import ro.unibuc.triplea.application.auth.dto.response.GameResponse;
+import ro.unibuc.triplea.domain.auth.service.GameService;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/v1/games")
+@RequiredArgsConstructor
+@Tag(name = "Game", description = "Game management APIs")
+public class GameController {
+    
+    private final GameService gameService;
+
+    @GetMapping("/get-all")
+    public ResponseEntity<List<GameResponse>> getAllGames(@RequestParam(required=false, name="count") Optional<Integer> count) {
+        List<GameResponse> games = gameService.getAllGames(count);
+        return ResponseEntity.ok(games);
+    }
+
+    @GetMapping("/{identifier}")
+    public ResponseEntity<?> getGameByIdentifier(@PathVariable String identifier) {
+        
+        Optional<GameResponse> game;
+
+        try {
```

Comment:
Business logic should be into service. Create it's own method in a service class.

Reply:
refactored code